kernel/init.c

```diff
@@ -424,6 +424,7 @@ static FUNC_NORETURN void switch_to_main_thread(void)
 }
 #endif /* CONFIG_MULTITHREADING */
 
+#if defined(CONFIG_STACK_CANARIES) || defined(CONFIG_STACK_POINTER_RANDOM)
```

Comment:
On second thought, noting how this function handles fake entropy device fallback as well, it should be:
```
#if defined(CONFIG_ENTROPY_HAS_DRIVER) || defined(CONFIG_TEST_RANDOM_GENERATOR)
```

p.s. The `random` subsystem configs seem to be somewhat of mess, maybe we should refactor it ...

Reply:
yeah, it is a mess and I'm looking for a complete overhaul in this subsystem. What you are suggesting was my first option since it is the STACK_POINTER_RANDOM dependency as well , but this function is problematic in many ways (duplicate code, assume some no blocking behaviors from random subsystem and uses `entropy_get_entropy_isr` without check how many bytes were written) that I thought would be better to limit its usage scope.

But yeah, will change the ifdef and solve these problems properly in other commit.